### DISCUSSION TITLE
New variable types

### DISCUSSION THREAD

**Discussion by arthuro555** [4ian/GDevelop#3018] [Enhancement ideas (and their technical discussions)]
I think 2 new data types for variables would be great:

### Set

A set just like the c++ standard library `std::set` or the JavaScript `Set`. An unordered set of values where a value can only be contained once. You could compare it to a structure with just the keys.

### Typed Arrays

JavaScript typed arrays are incredibly useful for everything performance related. It would be useful to load from the network or filesystem binary files and images. For example, this could allow to load an image from the internet via events or to use a binary-based serialization method like [FlatBuffers](https://google.github.io/flatbuffers/) to save games... 
They are especially needed since basic javascript optimizations are thrown out the window through having event number arrays contain objects (gdjs.Variables instead of numbers) 💀

**Comment by D8H** [comment]
I think that buffering is a technical concept that should be abstracted from users by event functions.
Why do you think it should be directly exposed to users?

**Reply by arthuro555** [reply]
Well buffer might not be the optimal term for that, what I am thinking about is more something like `Blob` so that one may download or load for example a binary file and write it/upload it somewhere else.

**Reply by 4ian** [reply]
Agree with D8H that this seems fairly far from a user need, and we would complexify the whole app for something not used a lot/at all. There is a very thin line between "overpowered" and "overcomplex" and I think we would cross it here.

**Reply by arthuro555** [reply]
I see. I just thought it might be more efficient than base64 strings. I often see people talk about wanting to load images from the web, firebase or the file system into their game and personally I think it might be useful to help towards that goal, but I can see your point.
